### PR TITLE
Search using GitHub link in /issues api

### PR DESCRIPTION
**Pull request by Ajeyakrishna-k**
Date: 4 November 2023
Developer Name: @Ajeyakrishna-k 

----

## Issue Ticket Number:- 
- #1591

## Description: 
Previously when pass a GitHub issue link in query string we were not getting an accurate response. This PR fixes that. 

Is Under Feature Flag 
- [x] Yes
- [ ] No

Database changes
- [ ] Yes
- [x] No

Breaking changes (If your feature is breaking/missing something please mention pending tickets)
- [ ] Yes
- [x] No

### Is Development Tested?

- [x] Yes
- [ ] No

Screenshot: 
Screenshot 2023-11-04 at 6 49 11 PM

Test code coverage: 
#### Code changes line numbers : [16 - 25]
Screenshot 2023-11-04 at 4 01 43 AM

### CODE REVIEW

**Comment by ajoykumardas12** [comment]
Looks good mostly. Some comments:
What happens if issue link matches regex but is invalid? i.e `/.+/` will match any repo name.
Also, any response screenshots of the feature?

File: controllers/issues.js [hunk 1]
```diff
@@ -10,9 +11,18 @@ const { SOMETHING_WENT_WRONG } = require("../constants/errorMessages");
...
+    const { q: queryString, dev } = req.query;
     let issues = {};
-    if (queryString) {
+    const githubOrg = config.get("githubApi.org");
+    const githubIssuerUrlPattern = new RegExp(`^https://github.com/${githubOrg}/.+/issues/\\d+$`);
+
+    if (githubIssuerUrlPattern.test(queryString) && dev === "true") {
+      const url = new URL(queryString);
+      const issueUrlPaths = url.pathname.split("/");
+      const repositoryName = issueUrlPaths[2];
```
Comment (iamitprakash) [inline controllers/issues.js:22]: what if issueUrlPaths has only 1 or 2 elements

**Review by iamitprakash — CHANGES_REQUESTED** [review]

**Review by bhtibrewal — COMMENTED** [review]
Please update the API contract and attach the PR for it here in the description

**Comment by Ajeyakrishna-k** [comment]
> What happens if issue link matches regex but is invalid? i.e `/.+/` will match any repo name. 

We will return an empty list.

File: controllers/issues.js (hunk 1 above)
Comment (Ajeyakrishna-k) [reply controllers/issues.js:22]: The ragex in [line 17](https://github.com/Real-Dev-Squad/website-backend/pull/1665/files/1d65902a4b30156b2cc817ed09e139aa4d57996f#diff-d0f9f68f392ba94b382d341a5f5651e2c8e06c368ff464d6d7d700bde9503273R17) will make sure we have required number of elements.

**Comment by Ajeyakrishna-k** [comment]
> Please update the API contract and attach the PR for it here in the description

This PR does not make any changes to DTOs.

**Comment by Ajeyakrishna-k** [comment]
> Also, any response screenshots of the feature?

Updated the description with the screenshot.

File: services/githubService.js [hunk 2]
```diff
@@ -281,7 +281,7 @@ const fetchIssuesById = async (repositoryName, issueId) => {
       logger.error(`GitHub API request failed. Status: ${res.status}, URL: ${url}`);
       return null;
     }
-    return res.json();
+    return await res.json();
```
Comment (heyrandhir) [inline services/githubService.js:284]: nit : return can be on next line.  It's unusual to see return and await on the same line

**Review by heyrandhir — APPROVED** [review]

**Review by Achintya-Chatterjee — APPROVED** [review]
:rocket:

**Review by ajoykumardas12 — APPROVED** [review]

**Review by iamitprakash — COMMENTED** [review]
j

**Review by iamitprakash — APPROVED** [review]

**Pull request merged**